rust/datafusion/src/execution/aggregate/max.rs

```diff
@@ -0,0 +1,179 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! MAX aggregate function
+
+use crate::arrow::array::*;
+use crate::arrow::compute;
+use crate::arrow::datatypes::DataType;
+
+use crate::error::{ExecutionError, Result};
+use crate::execution::aggregate::AggregateFunction;
+use crate::logicalplan::ScalarValue;
+
+/// Implementation of MAX aggregate function
+#[derive(Debug)]
+pub(super) struct MaxFunction {
+    data_type: DataType,
+    value: Option<ScalarValue>,
+}
+
+impl MaxFunction {
```

Comment:
Seems min, max and count are very similar. Is it possible to combine them into one, or reuse some of the common code?

Reply:
Perhaps a trait which implements the common functionality?